core/jvmMain/src/kotlinx/serialization/SerializersJvm.kt

```diff
@@ -74,6 +74,19 @@ public fun SerializersModule.serializer(type: Type): KSerializer<Any> =
 public fun SerializersModule.serializerOrNull(type: Type): KSerializer<Any>? =
     serializerByJavaTypeImpl(type, failOnMissingTypeArgSerializer = false)
 
+/**
+ * Retrieves a [KSerializer] for the given [java.lang.Class] instance of the Kotlin type or returns `null` if none is found.
+ * The given class must be annotated with [Serializable] or be one of the built-in types.
+ *
+ * The API avoids instantiation of the correspnding [KClass][Class.kotlin] on the best-effort basis.
+ *
+ * This is a [Class] counterpart of [KClass.serializerOrNull] with all the restrictions the original function implies,
+ * including the general recommendation to avoid uses of this API.
+ */
+@InternalSerializationApi
+public fun <T : Any> Class<T>.serializerOrNull(): KSerializer<T>? =
```

Comment:
Keep in mind that we already have `serializerOrNull` on `Type` that does exactly this and a bit of extra (e.g. checks for generics), I've just found out that making a separate overload is much easier from the maintenance standpoint than avoiding `KClass` instantiation on each path of `serializerOrNull(Type)`

Reply:
I'm still not sure we should have a separate overload, especially `@Internal` one. Maybe just rewrite `is Class<*>` branch in the implementation?

Reply:
That's a great point, I'll give it a try